internal/manifests/manifestutils/labels.go

```diff
@@ -18,16 +18,26 @@ import (
 	"regexp"
 	"strings"
 
+	"github.com/go-logr/logr"
+
 	"github.com/open-telemetry/opentelemetry-operator/apis/v1beta1"
 	"github.com/open-telemetry/opentelemetry-operator/internal/naming"
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+var logger logr.Logger
+
 func IsFilteredSet(sourceSet string, filterSet []string) bool {
-	for _, pattern := range filterSet {
-		match, _ := regexp.MatchString(pattern, sourceSet)
-		return match
+	for _, basePattern := range filterSet {
+		pattern, compileErr := regexp.Compile(basePattern)
+		match := pattern.MatchString(sourceSet)
+		if compileErr != nil {
+			logger.Error(compileErr, "could not compile the regexp pattern")
```

Comment:
is there a logger could pass in as a parameter instead here?

Reply:
I don't know if it's a good idea, once we call this function for each annotation/label we put in our resources.

Reply:
Sounds good. I've changed the function signature to support the logger we propagate.

Reply:
thank you for doing all that parameter drilling :)